### PR TITLE
Add Cart totals to Cart block

### LINKED ISSUE #1298: Cart Block: Full Cart view- Totals and Shipping sections - Edit Context
Note: when working on this issue, please give due consideration to:

- Are there components that can be pulled from `@wordpress/components` (i.e. `<Card>`)?
- Are there components that can be pulled from `@woocommerce/components`?
- Are there components/logic that can be pulled from [`@automattic/components`](https://github.com/Automattic/wp-calypso/tree/master/packages/components/src) or [`@automattic/composite-checkout`](https://github.com/Automattic/wp-calypso/tree/master/packages/composite-checkout)

We want to avoid duplication of work/logic already done elsewhere as much as possible.

As per the exploration done by @Aljullu in #1207 this issue is concerned with the totals and shipping areas of the Full Cart view:

full cart view

In particular, the totals and shipping sections:

Totals view

### Totals Components

* TotalsTable (`/base/components/totals/table`)
* TotalsTableItem (`/base/components/totals/table-item`)
* TotalsCouponCodeInput (`/base/components/totals/coupon-code-input`)

### Shipping Components

* ShippingOptionsList (`/base/components/shipping-options/list`)
* ShippingOptionsListItem (`/base/components/shipping-options/list-item`)

### PR DESCRIPTION
**Pull request by Aljullu**
Fixes #1298.

#### Accessibility

- [x] I've tested using only a keyboard (no mouse)
- [x] I've tested using a screen reader
- [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots

imatge

### How to test the changes in this Pull Request:

1. Create a post with a _Cart_ block.
2. Preview it in the frontend.
3. Verify it renders correctly and the button links to the checkout page.

### CODE REVIEW

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js [hunk 1]
```diff
@@ -8,6 +20,64 @@ import './style.scss';
...
+				'woo-gutenberg-products-block'
+			),
+			value: '€6.00',
+			description: __(
+				'Shipping to location (change address)',
+				'woo-gutenberg-products-block'
+			),
+		},
+	];
+	const shippingOptions = [
```
Comment (senadir) [inline assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Don’t we want to mimic the same behaviour we have at the checkout section?

File: bin/webpack-helpers.js [hunk 2]
```diff
@@ -124,6 +124,8 @@ const mainEntry = {
 	'active-filters': './assets/js/blocks/active-filters/index.js',
 	'block-error-boundary':
 		'./assets/js/base/components/block-error-boundary/style.scss',
+	'wordpress-components-style':
```
Comment (senadir) [inline bin/webpack-helpers.js]: why do we need this here?

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js (hunk 1 above)
Comment (Aljullu) [reply assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Absolutely! I think when I started working on this branch checkout was not yet merged, that's why I missed it.

I created a placeholder file in 8ccf09b3625ca859e982a5207922d2fb49ae27f6. We are displaying the options a little bit different depending on the screen, so I renamed the properties to make them easier to reuse in different situations.

File: bin/webpack-helpers.js (hunk 2 above)
Comment (Aljullu) [reply bin/webpack-helpers.js]: That's required to load `@wordpress/components` style so the `<Panel>` component looks as expected.

**Comment by Aljullu** [comment]
Thanks for the review @senadir. I applied and answered your feedback, feel free to take another look.

> also there is a small gray background?

I can't reproduce that. What browser are you using? Can you see this rule in the CSS `.wc-block-cart__shipping-options-fieldset { background-color: transparent }`?

File: bin/webpack-helpers.js (hunk 2 above)
Comment (nerrad) [reply bin/webpack-helpers.js]: > That's required to load @wordpress/components style so the <Panel> component looks as expected.

Will the [`@wordpress/base-styles`](https://www.npmjs.com/package/@wordpress/base-styles) package help with this?

Comment (Aljullu) [reply bin/webpack-helpers.js]: I don't think so. That package seems to include several SCSS variables used in Gutenberg styles, but in this case we want the `<Panel>` component styles, which are not included in that package.

**Comment by Aljullu** [comment]
After some more testing, I noticed that some styles from `@wordpress/components` were conflicting with our editor components:

imatge

In 69a141c0a08d8c16b4718ef534994084e8051b97 I refactored how styles are imported so instead of importing all styles from `@wordpress/components`, we only import the ones we need.

I'm marking this PR as `needs review` again so you can take another look.

**Comment by Aljullu** [comment]
I fixed the conflicts between this branch and `master` that appeared recently.

> I haven't reviewed anything, but just making a note that this must include this (if it isn't already):

@nerrad that's already included. :slightly_smiling_face: 

https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1411/files#diff-7357dea15a9376886f2447fd7b0ce974R105

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js [hunk 3]
```diff
@@ -16,6 +61,55 @@ const Cart = () => {
 				</span>
 			</div>
 			<div className="wc-block-cart__sidebar">
+				<h2 className="wc-block-cart__totals-title">
+					{ __( 'Cart totals', 'woo-gutenberg-products-block' ) }
+				</h2>
```
Comment (nerrad) [inline assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Does this follow the correct semantic structure for the cart block on the page (i.e. should it be `h3` or some other heading level)?

File: assets/js/blocks/cart-checkout/placeholder-shipping-methods.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+		schedule: '5 business days',
+	},
+	{
+		label: 'Express shipping',
+		value: 'ups-express',
+		dispatcher: 'Dispatched via USPS',
+		price: '€50.00',
+		schedule: '2 business days',
+	},
+];
```
Comment (nerrad) [inline assets/js/blocks/cart-checkout/placeholder-shipping-methods.js]: Does this follow the same shape as the response from the shipping endpoint? If not, it'd be a good idea to use the same shape and types as the rest endpoint response so it's easier to swap in the actual endpoint responses later.

File: bin/webpack-helpers.js [hunk 5]
```diff
@@ -192,6 +194,12 @@ const getMainConfig = ( options = {} ) => {
 						chunks: 'all',
 						priority: 10,
 					},
+					'vendors-style': {
+						test: /\/node_modules\/.*?style\.s?css$/,
+						name: 'vendors-style',
+						chunks: 'all',
+						priority: 7,
+					},
```
Comment (nerrad) [inline bin/webpack-helpers.js]: what is this configuration picking up?

File: assets/js/base/components/text-input/style.scss [hunk 6]
```diff
@@ -4,6 +4,7 @@
 }
 
 .wc-block-text-input label {
+	pointer-events: none;
```
[the remark this answers is not in the document]
Comment (Aljullu) [reply assets/js/base/components/text-input/style.scss]: Actually it was a wrong fix for another bug (a label missing its `for` attribute, so it was not clickable). Fixed it in 8e6fef97fe22723ca953419dfe1e1c3662e38ac4.

File: assets/js/base/components/totals/totals-coupon-code-input/index.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { PanelBody, PanelRow } from '@wordpress/components';
```
[the remark this answers is not in the document]
Comment (Aljullu) [reply assets/js/base/components/totals/totals-coupon-code-input/index.js]: It should, yes! I will update #1423 accordingly.

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js (hunk 3 above)
Comment (Aljullu) [reply assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Is this outlined somewhere? Looking at the designs, I thought _Cart_ would be the `h1` and _Shopping cart_ and _Cart totals_ would `h2`.

File: assets/js/blocks/cart-checkout/placeholder-shipping-methods.js (hunk 4 above)
Comment (Aljullu) [reply assets/js/blocks/cart-checkout/placeholder-shipping-methods.js]: It's not, but in #1468 I already replaced it so if it's ok I would leave it like it is in this pull and leave the update in #1468.

I will update the `totalItems` shape in this PR, though.

File: bin/webpack-helpers.js (hunk 5 above)
Comment (Aljullu) [reply bin/webpack-helpers.js]: The idea was to have `@wordpress/components` styles (and any future `node_modules` stylesheet) loaded in an independent file named `vendors-style.css`. But maybe it's better to bundle them in the generic `style.css`?

I think it all depends on how big the `vendors-style.css` file will end up being, now it's pretty small but as we keep adding more components from `@wordpress/components`, it will start getting bigger.

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js [hunk 8]
```diff
@@ -1,13 +1,58 @@
...
+				'Estimated shipping cost:',
+				'woo-gutenberg-products-block'
+			),
+			value: '€6.00',
+			description: __(
+				'Shipping to location (change address)',
+				'woo-gutenberg-products-block'
+			),
+		},
+	];
```
[the remark this answers is not in the document]
Comment (Aljullu) [reply assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Updated the shape in 4e8c5daeb0d4d99e54126013289ba14777936236.

**Comment by Aljullu** [comment]
Thanks for the review @nerrad! This is ready for another look.

File: assets/js/base/components/totals/totals-coupon-code-input/index.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (Aljullu) [reply assets/js/base/components/totals/totals-coupon-code-input/index.js]: Done in 570ec228af0e93ac067fb1db5716a1dcb7a194a4. :slightly_smiling_face:

File: assets/js/base/components/totals/totals-item/index.js [hunk 9]
```diff
@@ -3,19 +3,25 @@
...
 			className={ classnames( 'wc-block-totals-table-item', className ) }
 		>
 			<span className="wc-block-totals-table-item__label">{ label }</span>
-			<span className="wc-block-totals-table-item__value">{ value }</span>
+			<FormattedMonetaryAmount
+				className="wc-block-totals-table-item__value"
+				currency={ currency }
+				displayType="text"
+				value={ value }
+			/>
```
Comment (nerrad) [inline assets/js/base/components/totals/totals-item/index.js]: Noice 👍 - I forgot about this component that Mike had created earlier in the cycle - glad to see it used here!

File: assets/js/base/components/totals/totals-item/index.js [hunk 10]
```diff
@@ -24,8 +30,9 @@ const TotalsItem = ( { className, label, value, description } ) => {
 };
 
 TotalsItem.propTypes = {
+	currency: PropTypes.object.isRequired,
 	label: PropTypes.string.isRequired,
-	value: PropTypes.string.isRequired,
+	value: PropTypes.number.isRequired,
```
Comment (nerrad) [inline assets/js/base/components/totals/totals-item/index.js]: Hmm, are we sure this is a number? I _think_ values coming from the REST response are always a string.

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js [hunk 11]
```diff
@@ -26,28 +30,62 @@ const Cart = () => {
...
+		total_items: 6000,
+		total_items_tax: 0,
+		total_fees: 0,
+		total_fees_tax: 0,
+		total_discount: 0,
+		total_discount_tax: 0,
+		total_shipping: 0,
+		total_shipping_tax: 0,
+		total_tax: 0,
+		total_price: 6000,
```
Comment (nerrad) [inline assets/js/blocks/cart-checkout/cart/full-cart/index.js]: All these values coming from the rest request are of string type. So our sample data should be string too. We'll have to watch for the potential gotcha with `'0'` because it doesn't coerce to the same boolean value as `0`.

File: assets/js/blocks/cart-checkout/placeholder-shipping-methods.js (hunk 4 above)
Comment (nerrad) [reply assets/js/blocks/cart-checkout/placeholder-shipping-methods.js]: Yup works for me!

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js (hunk 3 above)
Comment (nerrad) [reply assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Oh this was more of a "gut check" comment to make sure it had been considered. From your reply it looks like you have 👍 and h2 is the correct semantics this spot in the rendered tree.

File: bin/webpack-helpers.js (hunk 5 above)
Comment (nerrad) [reply bin/webpack-helpers.js]: So is the idea here that there will be some blocks that don't need the vendors-style whereas _all_ blocks use the generic styles?  If so, then let's leave this separation in place for now.  We can always revisit if bundle needs change.

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js (hunk 11 above)
Comment (Aljullu) [reply assets/js/blocks/cart-checkout/cart/full-cart/index.js]: My bad, I changed it in the last moment thinking that was a bug in the API. :man_facepalming: 

Should be fixed in e687eafcbf6005aafee264e5b6c826c6093f367b. I moved the code that generates the table rows to a function to make it easier to read.

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js [hunk 12]
```diff
@@ -22,71 +22,92 @@ import placeholderShippingMethods from '../../placeholder-shipping-methods';
...
-		total_tax: 0,
-		total_price: 6000,
-	};
-
-	const currency = getCurrencyFromPriceResponse( cartTotals );
-	const totalItems = [
+const getTotalRows = ( cartTotals ) => {
+	const totalItems = parseInt( cartTotals.total_items );
+	const totalItemsTax = parseInt( cartTotals.total_items_tax );
+	const totalRows = [
```
Comment (nerrad) [inline assets/js/blocks/cart-checkout/cart/full-cart/index.js]: You are using `parseInt` for converting all the values in this new function.  Wherever, `parseInt` is being used should you also be including the `radix` value?  It doesn't default to `10` so we probably should be explicit here.

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js [hunk 13]
```diff
@@ -22,71 +22,92 @@ import placeholderShippingMethods from '../../placeholder-shipping-methods';
...
-				? cartTotals.total_fees + cartTotals.total_fees_tax
-				: cartTotals.total_fees,
+				? totalFees + totalFeesTax
+				: totalFees,
 		} );
 	}
-	if ( cartTotals.total_discount > 0 ) {
-		totalItems.push( {
+	const totalDiscount = parseInt( cartTotals.total_discount );
+	if ( parseInt( totalDiscount ) > 0 ) {
```
Comment (nerrad) [inline assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Does the `parseInt` need to be used here again when `totalDiscount` is already cast to int?

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js [hunk 14]
```diff
@@ -22,71 +22,92 @@ import placeholderShippingMethods from '../../placeholder-shipping-methods';
...
+		total_discount: '0',
+		total_discount_tax: '0',
+		total_shipping: '0',
+		total_shipping_tax: '0',
+		total_tax: '0',
+		total_price: '6000',
+	};
+
+	const currency = getCurrencyFromPriceResponse( cartTotals );
+	const totalRows = getTotalRows( cartTotals );
```
Comment (nerrad) [inline assets/js/blocks/cart-checkout/cart/full-cart/index.js]: The name for this value (and correspondingly it's function name) maybe could be tightened up a bit. Since this isn't the actual rendered rows, what about calling it `totalRowsConfig` or `totalRowsValues` or something like that instead?

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js [hunk 15]
```diff
@@ -22,71 +22,92 @@ import placeholderShippingMethods from '../../placeholder-shipping-methods';
...
+		total_items_tax: '0',
+		total_fees: '0',
+		total_fees_tax: '0',
+		total_discount: '0',
+		total_discount_tax: '0',
+		total_shipping: '0',
+		total_shipping_tax: '0',
+		total_tax: '0',
+		total_price: '6000',
+	};
```
Comment (nerrad) [inline assets/js/blocks/cart-checkout/cart/full-cart/index.js]: I think we should define this sample content outside of the component otherwise it's going to be a new object instance every time the function component is invoked (which will be a lot because of how react calls functional components).

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js [hunk 16]
```diff
@@ -148,7 +169,7 @@ const Cart = () => {
 					className="wc-block-cart__totals-footer"
 					currency={ currency }
 					label={ __( 'Total', 'woo-gutenberg-products-block' ) }
-					value={ cartTotals.total_price }
+					value={ parseInt( cartTotals.total_price ) }
```
Comment (nerrad) [inline assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Make sure you include the `radix` value for `parseInt` here.

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js (hunk 12 above)
Comment (Aljullu) [reply assets/js/blocks/cart-checkout/cart/full-cart/index.js]: I thought that shouldn't affect us since ES5 removed octal interpretation and defaults to decimal: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#ECMAScript_5_removes_octal_interpretation.

But yes, added the radix to make it explicit in 9755e7bca1c22c04522a737d744bde6f7abb7e89. Should we add an [eslint rule](https://eslint.org/docs/rules/radix) in a follow-up to make sure we are consistent with this?

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js (hunk 13 above)
Comment (Aljullu) [reply assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Absolutely not! Removed in 7b6dcac58175130f780080340aa4e6628e1921d0.

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js (hunk 14 above)
Comment (Aljullu) [reply assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Sounds good. Done in 8557012a7cf2453419bd2b1ad8b2f8c1983da78a.

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js (hunk 15 above)
Comment (Aljullu) [reply assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Well, that will go away once this component is hooked up to the API, but moved it out of the component anyway in 3c6c9c0e628cf3cfb70c7c94b8669713e648578c.

I moved `onActivateCoupon` out as well. We can move it inside in the future if it needs to read props or interact with the component state.

**Comment by Aljullu** [comment]
Thanks for the review @nerrad, this is ready for another pass.

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js (hunk 12 above)
Comment (nerrad) [reply assets/js/blocks/cart-checkout/cart/full-cart/index.js]: The MDN documentation has a [callout explicitly warning against assuming radix defaults to 10](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#Syntax).  While the documentation also references the ES5 standard, and our browser support range implements that standard, it still does not default to 10 and with that ambiguity I think it's important to be explicit. So thanks for making that change :)

> Should we add an eslint rule in a follow-up to make sure we are consistent with this?

Ya that's a good idea.

**Comment by Aljullu** [comment]
@nerrad I added three more commits in order to reduce the assets size (and get bundlewatch passing), will you have a second to take a look?

- 679ccee8a8db3b602175685ed8891f3b5947d48d splits the vendors assets to its own file (`vendors-frontend.js`).
- 3e159789200a51d5c0b0adc75d4ba25f1277bd88 updates `<Button>` to use our aliased version of `@wordpress/components`.
- 9f9b9759a98047b26e7d8f04189ffe78c1d5bb06 changes how we import `withRestApiHydration` so only that HOC is imported. I would like to evaluate how other components import it and make sure we are not loading unnecessary assets for them either.

This is the assets comparison:

| | Total JS assets size |
| --- | ---: |
| `master` | 6.17MB |
| This PR | 8.66MB |
| This PR after the last commits | 5.60MB |

CSS assets were analyzed here: https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1423#issuecomment-571532581.

File: bin/webpack-helpers.js [hunk 17]
```diff
@@ -331,6 +331,19 @@ const getFrontConfig = ( options = {} ) => {
 			// See https://webpack.js.org/configuration/output/#outputjsonpfunction
 			jsonpFunction: 'webpackWcBlocksJsonp',
 		},
+		optimization: {
```
Comment (senadir) [inline bin/webpack-helpers.js]: wondering what job this code does?

File: assets/js/blocks/cart-checkout/cart/frontend.js [hunk 18]
```diff
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { withRestApiHydration } from '@woocommerce/block-hocs';
```
Comment (senadir) [inline assets/js/blocks/cart-checkout/cart/frontend.js]: Not sure why was this necessary since we have side effects enabled, is it because the whole project is treated as a single package? would it be worth making each folder a separate package or just using relative paths?
cc @nerrad

File: src/BlockTypes/Cart.php [hunk 19]
```diff
@@ -48,6 +48,9 @@ public function render( $attributes = array(), $content = '' ) {
 			$this->block_name . '-frontend',
 			$this->block_name . '-block-frontend'
 		);
+		\Automattic\WooCommerce\Blocks\Assets::register_block_script(
+			'vendors-frontend'
+		);
```
Comment (nerrad) [inline src/BlockTypes/Cart.php]: Shouldn't this be enqueued before the block scripts?

Also, at some point (we could create an issue for now) we'll likely want to abstract this so that we have some way of automatically adding the vendors-frontend script as a dependency if needed (might be a way to do this for both `vendors` and `vendors-frontend` actually).

File: assets/js/blocks/cart-checkout/cart/frontend.js (hunk 18 above)
Comment (Aljullu) [reply assets/js/blocks/cart-checkout/cart/frontend.js]: It was including all HOCs imported from `index.js`, but after building it with `ENV='production'` it seems to work as expected, so false alarm. I reverted this change. Thanks for catching it @senadir & @nerrad.

File: assets/js/base/components/button/index.js [hunk 20]
```diff
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { Button as WPButton } from '@wordpress/components';
+import { Button as WPButton } from 'wordpress-components';
```
[the remark this answers is not in the document]
Comment (Aljullu) [reply assets/js/base/components/button/index.js]: > This means the direct import would be used in the editor as well (via our builds) right?

Yes, the consequence of loading these components from the aliased version is that we are loading `@wordpress/components` twice in the editor: the external and our aliased version.

That also means we are loading the styles twice:

imatge

That's far from the ideal solution, but I'm wary about only loading one stylesheet, because between different version one component might have been added or removed, so we can't reliably trust one of the stylesheets only.

A CSS-in-JS approach like the one used in the `<Card />` component, would solve that. But several components are not using it yet.

So technically there could be conflicts in the editor (not in the frontend, where we are only loading the aliased version), but I don't know how common that would be.

> we could probably introduce an alias that resolves to the direct import on frontend builds and an external on backend builds (via webpack config, since we separate out those builds), but I'm on the fence about whether it's overkill. If the first point above turns out not being an issue then this won't be.

That wouldn't work either because even in the editor we need to import the aliased version of `@wordpress/components` in some cases: for example when showing the _Cart_ block, which includes a `<Card />` component, we will always need the aliased version of `@wordpress/components`, no matter if it's in the frontend or the editor.

I can't think of any way to reliably load only one version of `@wordpress/components` in the editor, since the Gutenberg layout will be written based on one version that might be different from the version of our aliased package.

File: src/BlockTypes/Cart.php (hunk 19 above)
Comment (Aljullu) [reply src/BlockTypes/Cart.php]: > Shouldn't this be enqueued before the block scripts?

Sure! Actually I think it makes sense to define it as a dependency of the `cart-frontend.js` script. Done in 064a4532ab96624449ab9f90cdda97530253854e.

**Comment by Aljullu** [comment]
> Also, when we merge this, we should create an issue to apply these improvements to the All Products block as well for it's frontend build, we should get some size gains there as a result. That issue could then be planned for a future sprint (or maybe cooldown).

Considering the HOC issue was a [false alarm](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1411#discussion_r364321738) and _All Products_ doesn't load any component from `@wordpress/components`, I think none of these performance improvements can be applied to the _All Products_ block. Is that correct?

**Comment by senadir** [comment]
> Thanks for the review @senadir. I applied and answered your feedback, feel free to take another look.
> 
> > also there is a small gray background?
> 
> I can't reproduce that. What browser are you using? Can you see this rule in the CSS `.wc-block-cart__shipping-options-fieldset { background-color: transparent }`?

Yes, but it’s overruled by this

`.editor-styles-wrapper table:not( .has-background ) tbody tr:nth-child(2n) td, .editor-styles-wrapper fieldset, .editor-styles-wrapper fieldset legend`

**Comment by Aljullu** [comment]
> Yes, but it’s overruled by this

I could reproduce it now, thank you @senadir! Should be fixed in 9b142fad9021872d5d860a43aec5910ca105ba02.

> Doing another review, I think we should merge this, we can address any future surprises later, I do think we should open an issue about the double aliased dependency if we haven’t arealdy.

Sounds good, will merge this and any other issue we find can be fixed as a follow-up. I'm not sure what you mean by the "double aliased dependency issue", though.

**Comment by senadir** [comment]
@Aljullu according to the cart epic

> The user can show/hide the coupon code field. The coupon field should be on by default (it can be backlogged if we don't have enough time)

the coupon is off by default, any reason for that? changing it means setting this to true

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/base/components/totals/totals-coupon-code-input/index.js#L36

**Comment by Aljullu** [comment]
@senadir that was an overlook from my part. :sweat_smile:  Do you want to work on fixing this? Otherwise I would create a PR with a fix.
